mitiq/benchmarks/mirror_circuits.py

```diff
@@ -0,0 +1,134 @@
+    """Returns a circuit with a two-qubit Clifford gate applied to each edge in edges, and a random single-qubit
+    Clifford gate applied to every other qubit.
+
+    Args:
+        edges: A graph with the edges for which the two-qubit Clifford gate is to be applied.
+    """
+    gates = []
+    gates.append(cirq.CNOT(cirq.LineQubit(a), cirq.LineQubit(b)) for a,b in edges.edges)
+    circuit = cirq.Circuit(gates)
+    qubits = nx.Graph()
+    qubits.add_nodes_from(nx.isolates(edges))
+    circuit.append(random_single_cliffords(qubits))
+    return circuit
+
+
+def random_single_cliffords(qubits: nx.Graph) -> cirq.Circuit:
+    """Returns a circuit with a random single-qubit Clifford gate applied on each given qubit.
+    
+    Args:
+        qubits: A graph with each node representing a qubit for which a random single-qubit Clifford gate is to be applied.
+
+    """
+    gates = []
+    for qubit in qubits.nodes:
+        num = random.randint(len(cliffords))
+        gates.extend(clifford.on(cirq.LineQubit(qubit)) for clifford in cliffords[num])
+    return cirq.Circuit(gates)
+
+def generate_mirror_circuit(nlayers: int, two_qubit_gate_prob: float, connectivity_graph: nx.Graph) -> cirq.Circuit:
+    """Returns a randomized mirror circuit.
+    
+    Args:
+        nlayers: The number of random Clifford layers to be generated.
+        two_qubit_gate_prob: Probability of a two-qubit gate being applied.
+        connectivity_graph: The connectivity graph of the backend on which the mirror circuit will be run. This is
+            used to make sure 2-qubit gates are only run on connected qubits.
+      """
+    if not 0 <= two_qubit_gate_prob <= 1:
+        raise ValueError("two_qubit_gate_prob must be between 0 and 1")
+        
+    nqubits = connectivity_graph.number_of_nodes()
+    single_qubit_cliffords = random_single_cliffords(connectivity_graph)
+    
+    forward_circuit = cirq.Circuit()
+    
+    quasi_inversion_circuit = cirq.Circuit()
+    quasi_inverse_gates = []
+    
+    for _ in range(nlayers):
+        forward_circuit.append(random_paulis(nqubits))
+
+        selected_edges = edge_grab(two_qubit_gate_prob, connectivity_graph)
+        edge_graph = nx.Graph()
+        edge_graph.add_nodes_from(connectivity_graph)
+        edge_graph.add_edges_from(selected_edges.edges)
+        circ = random_cliffords(edge_graph)
+        forward_circuit.append(circ)
+        
+        quasi_inverse_gates.append(random_paulis(nqubits))
+        quasi_inverse_gates.append(cirq.inverse(circ))
+    
+    
+    quasi_inversion_circuit.append(gate for gate in reversed(quasi_inverse_gates))
+        
+    rand_paulis = cirq.Circuit()
+    rand_paulis.append(random_paulis(nqubits))
+    
+    circuit = single_qubit_cliffords + forward_circuit + rand_paulis + quasi_inversion_circuit + cirq.inverse(single_qubit_cliffords)
+    for x in range (nqubits):
+        circuit.append(cirq.measure(cirq.LineQubit(x)))
```

Comment:
Remove this and add a `cirq.measure(*cirq.LineQubit.range(nqubits))` to the `circuit = ` line. Besides being shorter, the multi-qubit `cirq.MeasurementGate` tends to be safer with conversions.